src/qibolab/instruments/erasynth.py

```diff
@@ -11,204 +11,111 @@
+        self.post("rfoutput", 0)
 
     @property
-    def power(self):
-        return self.settings.power
-
-    @power.setter
-    def power(self, x):
-        self.settings.power = x
-        if self.is_connected:
-            self._set_device_parameter("power", x)
-
-    def connect(self):
-        """Connects to the instrument using the IP address set in the runcard."""
-        if not self.is_connected:
-            for attempt in range(3):
-                try:
-                    if not self.ethernet:
-                        self.device = ERASynthPlusPlus(f"{self.name}", f"TCPIP::{self.address}::INSTR")
-                    else:
-                        self._post("readAll", 1)
-                        self._post("readDiagnostic", 0)
-                    self.is_connected = True
-                    break
-                except KeyError as exc:
-                    print(f"Unable to connect:\n{str(exc)}\nRetrying...")
-                    self.name += "_" + str(attempt)
-                except ConnectionError as exc:
-                    print(f"Unable to connect:\n{str(exc)}\nRetrying...")
-            if not self.is_connected:
-                raise InstrumentException(self, f"Unable to connect to {self.name}")
-        else:
-            raise RuntimeError("There is an open connection to the instrument already")
-        # set proper frequency and power if they were changed before connecting
-        if self.settings.frequency is not None:
-            self._set_device_parameter("frequency", self.settings.frequency)
-        if self.settings.power is not None:
-            self._set_device_parameter("power", self.settings.power)
-
-    def _set_device_parameter(self, parameter: str, value):
-        """Sets a parameter of the instrument, if it changed from the last stored in the cache.
-
-        Args:
-            parameter: str = The parameter to be cached and set.
-            value = The value to set the paramter.
-        Raises:
-            Exception = If attempting to set a parameter without a connection to the instrument.
-        """
-        if not (parameter in self._device_parameters and self._device_parameters[parameter] == value):
-            if self.is_connected:
-                if not self.ethernet:
-                    if not hasattr(self.device, parameter):
-                        raise ValueError(f"The instrument {self.name} does not have parameter {parameter}")
-                    self.device.set(parameter, value)
-                else:
-                    if parameter == "power":
-                        self._post("amplitude", float(value))
-                    elif parameter == "frequency":
-                        self._post("frequency", int(value))
-                self._device_parameters[parameter] = value
-            else:
-                raise ConnectionError(f"Attempting to set {parameter} without a connection to the instrument")
+    def url(self):
+        return f"http://{self.address}/"
```

Comment:
I wonder whether this is even more general (how much we could lift up to the parent classes).

Reply:
I'd prefer to keep here for now, until it appears somewhere else, because I don't think it will.